components/operator/internal/chart/flags.go

```diff
@@ -32,12 +33,41 @@ func NewFlagsBuilder() FlagsBuilder {
 func (fb *flagsBuilder) Build() map[string]interface{} {
 	flags := map[string]interface{}{}
 	for key, value := range fb.flags {
-		flagPath := strings.Split(key, ".")
+		flagPath := strings.FieldsFunc(key, fieldsFuncWithExtrudes(key))
+		flagPath = removeExcludesFromFlagPath(flagPath)
 		appendFlag(flags, flagPath, value)
 	}
 	return flags
 }
 
+func fieldsFuncWithExtrudes(flag string) func(rune) bool {
+	index := 0
+	return func(r rune) bool {
+		split := shouldBeSplit(flag, index, r)
+
+		// increase index to know on which rune we are right now
+		index++
+
+		// return false if
+		return split
+	}
+}
+
+func shouldBeSplit(flag string, index int, r rune) bool {
+	if r != '.' {
+		return false
+	}
+
+	return index > 0 && flag[index-1] != '\\'
+}
+
+func removeExcludesFromFlagPath(flagPath []string) []string {
```

Comment:
`\\` aren't excludes but escapes, I'd change the name here to something like `removeEscapesFromFlagPath`